rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/BaseTransactionalExecutor.java

```diff
@@ -201,15 +201,12 @@ protected String buildLockKey(TableRecords rowsIncludingPK) {
         StringBuilder sb = new StringBuilder();
         sb.append(rowsIncludingPK.getTableMeta().getTableName());
         sb.append(":");
-
-        boolean flag = false;
+        int filedSequence = 0;
         for (Field field : rowsIncludingPK.pkRows()) {
-            if (flag) {
+            sb.append(field.getValue());
+            if (++filedSequence < rowsIncludingPK.pkRows().size()) {
```

Comment:
```java
sb.append(field.getValue());
filedSequence++;
 if (filedSequence < rowsIncludingPK.pkRows().size()) {
```
Better to not lead the if statement too complex.

Reply:
accepted